packages/core/src/network/network.ts

```diff
@@ -0,0 +1,435 @@
+              });
+            });
+
+            // Format the results with agent names for clarity
+            return actions.map((action, index) => `[${action.agent}]: ${results[index]}`).join('\n\n');
+          } catch (err) {
+            // Properly type the error
+            const error = err as Error;
+            console.error('Error in transmit tool:', error);
+            return `Error executing agents: ${error.message}`;
+          }
+        },
+      }),
+    };
+  }
+
+  #addToAgentHistory(agentId: string, interaction: { input: string; output: string }) {
+    if (!this.#agentHistory[agentId]) {
+      this.#agentHistory[agentId] = [];
+    }
+    // Add timestamp to the interaction
+    this.#agentHistory[agentId].push({
+      ...interaction,
+      timestamp: new Date().toISOString(),
+    });
+  }
+
+  getAgentHistory(agentId: string) {
+    return this.#agentHistory[agentId] || [];
+  }
+
+  #clearNetworkHistoryBeforeRun() {
+    this.#agentHistory = {};
+  }
+
+  /**
+   * Get the history of all agent interactions that have occurred in this network
+   * @returns A record of agent interactions, keyed by agent ID
+   */
+  getAgentInteractionHistory() {
+    return { ...this.#agentHistory };
+  }
+
+  /**
+   * Get a summary of agent interactions in a more readable format, displayed chronologically
+   * @returns A formatted string with all agent interactions in chronological order
+   */
+  getAgentInteractionSummary() {
+    const history = this.#agentHistory;
+    const agentIds = Object.keys(history);
+
+    if (agentIds.length === 0) {
+      return 'No agent interactions have occurred yet.';
+    }
+
+    // Collect all interactions with their agent IDs
+    const allInteractions: Array<{
+      agentId: string;
+      interaction: { input: string; output: string; timestamp: string };
+      index: number;
+      // We'll use a global sequence number to track the true chronological order
+      sequence: number;
+    }> = [];
+
+    // Track the global sequence of interactions
+    let globalSequence = 0;
+
+    // Gather all interactions with their source agent
+    agentIds.forEach(agentId => {
+      const interactions = history[agentId] || [];
+      interactions.forEach((interaction, index) => {
+        allInteractions.push({
+          agentId,
+          interaction,
+          index,
+          // Assign a sequence number based on when it was added to the history
+          sequence: globalSequence++,
+        });
+      });
+    });
+
+    // Sort by timestamp for true chronological order
+    allInteractions.sort((a, b) => {
+      // Compare timestamps if available
+      if (a.interaction.timestamp && b.interaction.timestamp) {
+        return new Date(a.interaction.timestamp).getTime() - new Date(b.interaction.timestamp).getTime();
+      }
+      // Fall back to sequence number if timestamps aren't available
+      return a.sequence - b.sequence;
+    });
```

Comment:
Without a threads concept I think this means agent network is single tenant + single usecase right now. Multiple workflows or multiple different chats using the same network will share interaction histories right? Using memory would fix it, but we could also do an in-memory version of this without actual Memory by having them pass a resourceId+threadId and storing it in a map on the class instance `Map<resourceId, Map<threadId, interactions>>`

Reply:
vNext for sure